### PR TITLE
fix: resolve azure-hosted-copilot-sdk test failures for timeout and BYOM API key leak (#1447, #1448)

### LINKED ISSUE #1447: Integration test failure: azure-hosted-copilot-sdk – greenfield copilot SDK explicit mention [Timeout]
## Prompt

```
Build an Azure app that uses the Copilot SDK to brutally review GitHub repos based on user input
```

## Summary

**Run:** [Integration Tests - all #110](https://github.com/microsoft/GitHub-Copilot-for-Azure/actions/runs/23379277272)
**Test:** `azure-hosted-copilot-sdk_ - Integration Tests › skill-invocation › greenfield: invokes skill when prompt mentions copilot SDK`
**Result:** Fail
**Duration:** 20m 1s (timeout)

## Root Cause Category

**Timeout**

### What was expected
The test calls `measureInvocationRate` (3 runs) and expects the `azure-hosted-copilot-sdk` skill to be invoked at ≥ 60% (0.6) when the prompt explicitly mentions "Copilot SDK". The `TEST_TIMEOUT` is 20 minutes.

### What actually happened
Jest threw: `"Exceeded timeout of 1200000 ms for a test."` — the entire `measureInvocationRate` loop (3 agent runs) exceeded the 20-minute budget. At least one individual agent run must have been very slow.

### Why it failed
Greenfield + explicit Copilot SDK prompt triggers full planning + code generation. With `RUNS_PER_PROMPT` = 3 and slow model responses, the cumulative runtime exceeds 20 minutes. This is a recurring pattern for codebase-scan tests in this suite (see also #1418).

### Suggested fix
- Reduce `RUNS_PER_PROMPT` to 1 or 2 for this specific greenfield scenario
- Increase `TEST_TIMEOUT` for the greenfield-explicit test or move it to a separate slow-running suite

## azure-hosted-copilot-sdk Skill Invocation

| Skill | Invoked |
|-------|---------|
| **azure-hosted-copilot-sdk** | Unknown (timed out before invocation rate was measured) |

## Skill Report Context

Skill report artifact not accessible (network policy blocks Azure Blob Storage downloads in the AWF sandbox). See run artifacts at https://github.com/microsoft/GitHub-Copilot-for-Azure/actions/runs/23379277272/artifacts/6039710044

## Environment

- **Runner OS:** ubuntu-latest
- **Node.js:** v24.x
- **Run URL:** https://github.com/microsoft/GitHub-Copilot-for-Azure/actions/runs/23379277272
- **Commit:** `2f186f53859a62d458f62897cf089476ca1eb05d`
- **Test file:** `azure-hosted-copilot-sdk/integration.test.ts:57`

> Generated by [Analyze Test Run](https://github.com/microsoft/GitHub-Copilot-for-Azure/actions/runs/23381600951) · [◷](https://github.com/search?q=repo%3Amicrosoft%2FGitHub-Copilot-for-Azure+is%3Aissue+%22gh-aw-workflow-call-id%3A+microsoft%2FGitHub-Copilot-for-Azure%2Fanalyze-test-run%22&type=issues)

> [!WARNING]
> <details>
> <summary>⚠️ Firewall blocked 1 domain</summary>
>
> The following domain was blocked by the firewall during workflow execution:
>
> - `productionresultssa3.blob.core.windows.net`
>
> To allow these domains, add them to the `network.allowed` list in your workflow frontmatter:
>
> ```yaml
> network:
>   allowed:
>     - defaults
>     - "productionresultssa3.blob.core.windows.net"
> ```
>
> See [Network Configuration](https://github.github.com/gh-aw/reference/network/) for more information.
>
> </details>

### LINKED ISSUE #1448: Integration test failure: azure-hosted-copilot-sdk – BYOM config API key leak [Assertion mismatch]
## Prompt

BYOM (Bring Your Own Model) configuration scenario — agent is asked to configure a Copilot SDK app to use a custom model endpoint.

## Summary

**Run:** [Integration Tests - all #110](https://github.com/microsoft/GitHub-Copilot-for-Azure/actions/runs/23379277272)
**Test:** `azure-hosted-copilot-sdk_ - Integration Tests › content-quality › BYOM prompt mentions DefaultAzureCredential`
**Result:** Fail
**Duration:** 4m 43s

## Root Cause Category

**Assertion mismatch**

### What was expected
The test runs a regression detector (`countApiKeyInByomConfig`) on raw agent metadata to verify that the agent uses `DefaultAzureCredential`-based auth (managed identity) rather than static API keys when generating BYOM configuration. Expected count: `0`.

### What actually happened
`countApiKeyInByomConfig(rawMetadata)` returned `2` — the agent included API key references twice in the generated BYOM configuration instead of recommending `DefaultAzureCredential`.

### Why it failed
The agent's response to the BYOM configuration prompt included API key patterns (e.g. `api_key`, `apiKey`, or similar) in the generated SDK config, rather than using managed identity auth. This is a content quality regression — the skill should guide users toward credential-free authentication.

### Suggested fix
- Update `azure-hosted-copilot-sdk` skill instructions to explicitly guide toward `DefaultAzureCredential` for BYOM scenarios
- Add or strengthen the "no static API keys in BYOM config" requirement in the skill's SKILL.md
- Check if a recent model update changed how BYOM config is suggested

## azure-hosted-copilot-sdk Skill Invocation

| Skill | Invoked |
|-------|---------|
| **azure-hosted-copilot-sdk** | N/A (content-quality test, not a skill-invocation test) |

## Skill Report Context

Skill report artifact not accessible (network policy blocks Azure Blob Storage downloads in the AWF sandbox). See run artifacts at https://github.com/microsoft/GitHub-Copilot-for-Azure/actions/runs/23379277272/artifacts/6039710044

## Environment

- **Runner OS:** ubuntu-latest
- **Node.js:** v24.x
- **Run URL:** https://github.com/microsoft/GitHub-Copilot-for-Azure/actions/runs/23379277272
- **Commit:** `2f186f53859a62d458f62897cf089476ca1eb05d`
- **Test file:** `azure-hosted-copilot-sdk/integration.test.ts:128`

> Generated by [Analyze Test Run](https://github.com/microsoft/GitHub-Copilot-for-Azure/actions/runs/23381600951) · [◷](https://github.com/search?q=repo%3Amicrosoft%2FGitHub-Copilot-for-Azure+is%3Aissue+%22gh-aw-workflow-call-id%3A+microsoft%2FGitHub-Copilot-for-Azure%2Fanalyze-test-run%22&type=issues)

> [!WARNING]
> <details>
> <summary>⚠️ Firewall blocked 1 domain</summary>
>
> The following domain was blocked by the firewall during workflow execution:
>
> - `productionresultssa3.blob.core.windows.net`
>
> To allow these domains, add them to the `network.allowed` list in your workflow frontmatter:
>
> ```yaml
> network:
>   allowed:
>     - defaults
>     - "productionresultssa3.blob.core.windows.net"
> ```
>
> See [Network Configuration](https://github.github.com/gh-aw/reference/network/) for more information.
>
> </details>

### PR DESCRIPTION
**Pull request by jongio**
Fixes #1447
Fixes #1448

## Summary

Two fixes for `azure-hosted-copilot-sdk` integration test failures.

### Issue #1447: Greenfield test timeout

The greenfield-explicit test ran `measureInvocationRate` with 3 runs. Each greenfield invocation triggers full planning + code scaffolding (~6-8 min), exceeding the 20-min `TEST_TIMEOUT`.

**Fix:** Introduced `GREENFIELD_RUNS_PER_PROMPT = 1` for the greenfield-explicit scenario, matching the pattern used by other slow suites (azure-deploy, azure-prepare). With 1 run and the 0.6 threshold, the test requires 100% success rate (1/1 = 1.0 >= 0.6), appropriate for an explicit SDK mention.

### Issue #1448: BYOM config API key leak

`countApiKeyInByomConfig(rawMetadata)` returned 2 instead of expected 0. The SKILL.md lacked explicit prohibition of `apiKey` patterns in BYOM configuration.

**Fix:** Added a prominent BYOM Auth callout in Step 3 requiring `DefaultAzureCredential` / `ManagedIdentityCredential` and explicitly prohibiting `apiKey`, `AZURE_OPENAI_API_KEY`, and `AZURE_OPENAI_KEY`. Added matching rule in Rules section. Bumped version 1.0.3 -> 1.0.4.

## Changes

| File | Change |
|------|--------|
| `tests/.../integration.test.ts` | `GREENFIELD_RUNS_PER_PROMPT = 1` for slow greenfield test |
| `plugin/skills/.../SKILL.md` | BYOM auth guidance + version bump to 1.0.4 |
| `tests/.../__snapshots__/triggers.test.ts.snap` | New `identity` keyword from `ManagedIdentityCredential` |

## Testing

- 42/42 unit + trigger tests pass
- No lint/type errors or warnings
- MQ quality gate passed (15-step, 4-wave pipeline)

### CODE REVIEW

File: tests/azure-hosted-copilot-sdk/integration.test.ts [hunk 1]
```diff
@@ -38,6 +38,11 @@ import {
 
 const SKILL_NAME = "azure-hosted-copilot-sdk";
 const RUNS_PER_PROMPT = 3;
+// Greenfield-explicit prompts trigger full planning + scaffolding per run (~6-8 min each).
+// Three runs exceed the 20-min timeout. One run is sufficient to validate routing correctness
+// for an explicit SDK mention, matching the pattern used by other slow suites (azure-deploy,
+// azure-prepare). See: https://github.com/microsoft/GitHub-Copilot-for-Azure/issues/1447
+const GREENFIELD_RUNS_PER_PROMPT = 1;
```
[the remark this answers is not in the document]
Comment (jongio) [reply tests/azure-hosted-copilot-sdk/integration.test.ts:45]: The 4-line comment block directly above the constant already specifies this is for `greenfield-explicit prompts.` The constant is used exactly once (line 71) inside the only greenfield test. There are no other greenfield scenarios that could cause confusion — the remaining tests are `existing-app-deploy`, `existing-app-modify`, and `byom-config`. Adding `_EXPLICIT_` to a test-internal constant adds verbosity for zero practical disambiguation.

File: plugin/skills/azure-hosted-copilot-sdk/SKILL.md [hunk 2]
```diff
@@ -56,6 +56,8 @@ Three model paths (layers on top of 2A/2B):
 | **GitHub specific** | `model: "<name>"` — use `listModels()` to discover |
 | **Azure BYOM** | `model` + `provider` with `bearerToken` via `DefaultAzureCredential` |
 
+> ⚠️ **BYOM Auth — MANDATORY**: Azure BYOM configurations MUST use `DefaultAzureCredential` (local dev) or `ManagedIdentityCredential` (production) to obtain a `bearerToken`. **NEVER** use static API keys (`apiKey`), `AZURE_OPENAI_API_KEY`, or `AZURE_OPENAI_KEY` environment variables in BYOM provider configuration. See [auth-best-practices.md](references/auth-best-practices.md) for the credential pattern and [model config ref](references/azure-model-config.md) for the full BYOM code example.
+
```
[the remark this answers is not in the document]
Comment (jongio) [reply plugin/skills/azure-hosted-copilot-sdk/SKILL.md:60]: These are not contradictory. `DefaultAzureCredential` is a credential chain that includes `ManagedIdentityCredential` as one of its providers — they have a general/specific relationship, not a conflicting one. The table row (pre-existing, not part of this PR) correctly uses `DefaultAzureCredential` as the quick-reference summary. The callout we added provides supplementary production guidance for teams that want explicit `ManagedIdentityCredential`. The two work together by design.

**Review by kvenkatrajan — APPROVED** [review]
